jwst/group_scale/group_scale.py

```
@@ -40,7 +40,17 @@ def do_correction(model):
 
     # Apply the rescaling to the entire data array
     scale = float(frame_divisor) / nframes
-    model.data *= scale
+    try:
+        # Multiply in place if possible
+        model.data *= scale
+    except TypeError:
```

Hmm why wait till it crash? Is it not possible to check dtype?

---

I suppose I could?  I was thinking that it might be a number of possible float types or int types, and it's only a problem if it doesn't cast correctly, but I could check the dtype for the expected failure case (np.uint16, for a Level1bModel) and let it fail the old ugly way if it's not that one.

---

Better? [e66bb4b](https://github.com/spacetelescope/jwst/pull/9903/commits/e66bb4be92cee60c483ec6883c92a4edec158cac)

---

Agreed, and thanks for pointing it out!